src/dbms/dbms_handler.cpp

```diff
@@ -275,18 +261,50 @@ DbmsHandler::DeleteResult DbmsHandler::TryDelete(std::string_view db_name) {
 
   // Delete disk storage
   std::error_code ec;
-  (void)std::filesystem::remove_all(*storage_path, ec);
+  (void)std::filesystem::remove_all(storage_path, ec);
   if (ec) {
-    spdlog::error(R"(Failed to clean disk while deleting database "{}" stored in {})", db_name, *storage_path);
+    spdlog::error(R"(Failed to clean disk while deleting database "{}" stored in {})", db_name, storage_path);
   }
-  return {};  // Success
+
+  // Success
+  // Save delta
+  if (system_transaction_) {
+    system_transaction_->delta.emplace(SystemTransaction::Delta::drop_database, uuid);
+  }
+  return {};
 }
 
 DbmsHandler::DeleteResult DbmsHandler::Delete(std::string_view db_name) {
   auto wr = std::lock_guard(lock_);
   return Delete_(db_name);
 }
 
+DbmsHandler::DeleteResult DbmsHandler::Delete(utils::UUID uuid) {
+  auto wr = std::lock_guard(lock_);
+  std::string db_name;
+  try {
+    const auto db = Get_(uuid);
+    db_name = db->name();
+  } catch (const UnknownDatabaseException &) {
+    return DeleteError::NON_EXISTENT;
+  }
+  return Delete_(db_name);
```

Comment:
TODO: new overload of Delete_ that works with db we already have (no need to re-Get by name)